pythonx/jedi_vim.py

```diff
@@ -716,6 +716,10 @@ def do_rename(replace, orig=None):
         # Save view.
         saved_view = vim_eval('string(winsaveview())')
 
+        # Open all folds in the window, so that replacing of original
+        # word doesn't overwrite the folded section.
+        vim_command('normal! zR')
+
         # Replace original word.
         vim.current.window.cursor = (r.line, r.column)
         vim_command('normal! c{0:d}l{1}'.format(len(orig), replace))
```

Comment:
I think we should use another API to change the word instead, likely the `setline()` function.
This avoids (re)storing the view also.

Reply:
You could just set `vim.current.buffer[r.line - 1]` to the resulting string (where the replacement was done in/via Python).

Reply:
oh nice, thanks! should be solved now